source/time_stepping/convection_time_step.cc

```diff
@@ -36,13 +36,19 @@ namespace aspect
       FEValues<dim> fe_values (this->get_mapping(),
                                this->get_fe(),
                                quadrature_formula,
-                               update_values);
+                               update_values   |
+                               update_gradients |
+                               update_quadrature_points |
+                               update_JxW_values);
```

Comment:
Only compute these properties if needed, it slows down the computation of the timestep significantly. Check [here](https://github.com/geodynamics/aspect/blob/0d485a6bb605e2f4d753b37a9de3a760b8e4f924/source/simulator/assembly.cc#L736) for how to do this.